vsutil/__init__.py

```diff
@@ -35,6 +37,47 @@ class Dither(Enum):
     ERROR_DIFFUSION = 'error_diffusion'  # Floyd-Steinberg error diffusion.
 
 
+def is_variable(clip: vs.VideoNode, /, *, format: bool = False, resolution: bool = False) -> bool:
+    """
+    Returns True if at least one of the specified clip's attributes is variable.
+    It is an error to use this without specifying which attribute to check.
+    """
+    if not format and not resolution:
+        raise ValueError('At least one attribute must be specified as `True`.')
+
+    if format:
+        if clip.format is None:
+            return True
+    if resolution:
+        if 0 in (clip.width, clip.height):
+            return True
+
+    return False
+
+
+def disallow_variable_format(function: Callable[..., R]) -> Callable[..., R]:
+    """
+    Function decorator that raises an exception if the input clip has a variable format.
+    """
+    def _check(clip: vs.VideoNode, *args, **kwargs) -> R:
```

Comment:
Add

```py
@functools.wraps(function)
```

to the wrapper function (`_check`).

Reply:
Added in https://github.com/Irrational-Encoding-Wizardry/vsutil/pull/37/commits/f942ce28bc67a88992a87409adc07f55c036c99a, is this so `function.__name__` and `function.__doc__` still work?